File: src/runtime/boxes/BbFail.java
/**
 * BbFail.java — FAIL: always ω — force backtrack
 * Port of bb_fail.c / bb_fail.s
 */
class BbFail extends BbBox {

    public BbFail(MatchState ms) { super(ms); }

    @Override public Spec alpha() { return null; }
    @Override public Spec beta()  { return null; }
}

File: src/runtime/boxes/bb_fail.c
/* _XFAIL    FAIL        always ω — force backtrack */
#include "bb_box.h"
#include <stdlib.h>
#include <string.h>


spec_t bb_fail(void *zeta, int entry)
{
    (void)zeta; (void)entry;
    return spec_empty;
}

fail_t *bb_fail_new(void)
{ return calloc(1,sizeof(fail_t)); }
